pkg/apply/solver/solver.go

```diff
@@ -119,19 +119,48 @@ func (t *TaskQueueBuilder) WithPruneObjects(pruneObjs object.UnstructuredSet) *T
 }
 
 // Build returns the queue of tasks that have been created
-func (t *TaskQueueBuilder) Build(o Options) *TaskQueue {
+func (t *TaskQueueBuilder) Build(taskContext *taskrunner.TaskContext, o Options) *TaskQueue {
 	var tasks []taskrunner.Task
 
 	// reset counters
 	t.applyCounter = 0
 	t.pruneCounter = 0
 	t.waitCounter = 0
 
+	// Filter objects that failed earlier validation
 	applyObjs := t.Collector.FilterInvalidObjects(t.applyObjs)
 	pruneObjs := t.Collector.FilterInvalidObjects(t.pruneObjs)
 
+	// Merge applyObjs & pruneObjs and graph them together.
+	// This detects implicit and explicit dependencies.
+	// Invalid dependency annotations will be treated as validation errors.
+	allObjs := make(object.UnstructuredSet, 0, len(applyObjs)+len(pruneObjs))
+	allObjs = append(allObjs, applyObjs...)
+	allObjs = append(allObjs, pruneObjs...)
+	g, err := graph.DependencyGraph(allObjs)
+	if err != nil {
+		t.Collector.Collect(err)
+	}
+	// Store graph for use by DependencyFilter
+	taskContext.SetGraph(g)
+	// Sort objects into phases (apply order).
+	// Cycles will be treated as validation errors.
+	idSetList, err := g.Sort()
+	if err != nil {
+		t.Collector.Collect(err)
```

Comment:
Does `Collect(err)` send an error event?

Reply:
Nope. It just collects the errors. The applier and destroyer have the code to either send the collected errors as error events or return as an error.

Reply:
Ack.